blockly-core/core/ui/block.js

```diff
@@ -1542,12 +1542,19 @@ Blockly.Block.prototype.setCurrentlyHidden = function (hidden) {
  * CurrentlyHidden is a non-persistent property used to hide certain blocks
  * (like function definitions/examples) that should only be visible when using
  * the modal function editor.
- * This method calculates whether this block is currently visible
+ * This method calculates whether this block is currently visible based
+ * on our current block-editing state (Blockly.editBlocks)
  * @returns true if both visibility conditions are met.
  */
 Blockly.Block.prototype.isVisible = function () {
   var visibleThroughParent = !this.parentBlock_ || this.parentBlock_.isVisible();
-  return visibleThroughParent && this.isUserVisible() && !this.isCurrentlyHidden_();
+  var visible = visibleThroughParent && !this.isCurrentlyHidden_();
+
+  if (!Blockly.editBlocks) {
+    visible = visible && this.isUserVisible();
+  }
```

Comment:
I might rewrite this section as
```
if (Blockly.editBlocks) {
  // We want to show this blocks whether or not isUserVisible is true
  return visible;
}

return visible && this.isUserVisible();
```
I thinks it's easier to wrap my head around without the negative.